// @react-spring/shared/helpers.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = require("react");
exports.is = {
    arr: Array.isArray,
    obj: function (a) {
        return !!a && a.constructor.name === 'Object';
    },
    fun: function (a) { return typeof a === 'function'; },
    str: function (a) { return typeof a === 'string'; },
    num: function (a) { return typeof a === 'number'; },
    und: function (a) { return a === void 0; },
    boo: function (a) { return typeof a === 'boolean'; },
};
function useOnce(effect) {
    react_1.useEffect(effect, []);
}
exports.useOnce = useOnce;
function useForceUpdate() {
    var _a = react_1.useState(false), f = _a[1];
    var forceUpdate = react_1.useCallback(function () { return f(function (v) { return !v; }); }, []);
    return forceUpdate;
}
exports.useForceUpdate = useForceUpdate;
/** Use a value from the previous render */
function usePrev(value) {
    var prevRef = react_1.useRef(undefined);
    var prev = prevRef.current;
    prevRef.current = value;
    return prev;
}
exports.usePrev = usePrev;
